code/goonstation/modules/power/power_transmission_laser/transmission_laser.dm

```diff
@@ -0,0 +1,448 @@
+				charge += input_pulling
+			else
+				inputting = FALSE
+		else
+			if(input_attempt && input_pulling > 0)
+				inputting = TRUE
+	else
+		inputting = FALSE
+
+	if(charge < MINIMUM_POWER)
+		firing = FALSE
+		output_level = 0
+		destroy_lasers()
+		return
+
+	if(!firing)
+		return
+
+	output_level = min(charge, output_number * power_format_multi_output)
+	if(!length(laser_effects))
+		setup_lasers()
+
+	if(length(blocked_objects))
+		var/atom/listed_atom = blocked_objects[1]
+		if(prob(max((abs(output_level) * 0.1) / 500 KW, 1))) ///increased by a single % per 500 KW
+			listed_atom.visible_message("<span class='danger'>[listed_atom] is melted away by the [src]!<span/>")
+			blocked_objects -= listed_atom
+			qdel(listed_atom)
+
+	else
+		sell_power(output_level * WATT_TICK_TO_JOULE)
+
+	charge -= output_level
+
+//// Selling defines are here
+#define MINIMUM_BAR 1
+#define PROCESS_CAP (20 - MINIMUM_BAR)
+
+#define A1_CURVE 20
+
+/obj/machinery/power/transmission_laser/proc/sell_power(joules)
+	var/mega_joules = joules / (1 MW)
+
+	var/generated_cash = (2 * mega_joules * PROCESS_CAP) / ((2 * mega_joules) + (PROCESS_CAP * A1_CURVE))
+	generated_cash += (4 * mega_joules * MINIMUM_BAR) / (4 * mega_joules + MINIMUM_BAR)
+	generated_cash = round(generated_cash)
+	if(generated_cash < 0)
+		return
+
+	total_energy += mega_joules
+	total_earnings += generated_cash
+	generated_cash += unsent_earnings
+	unsent_earnings = generated_cash
+
+	var/datum/money_account/engineering_bank_account = GLOB.station_money_database.get_account_by_department(DEPARTMENT_ENGINEERING)
+	var/datum/money_account/cargo_bank_account = GLOB.station_money_database.get_account_by_department(DEPARTMENT_SUPPLY)
+
+	var/medium_cut = generated_cash * 0.25
+	var/high_cut = generated_cash * 0.75
+
+	cargo_bank_account.deposit_credits(medium_cut, "Transmission Laser Payout")
+	unsent_earnings -= medium_cut
+
+	engineering_bank_account.deposit_credits(high_cut, "Transmission Laser Payout")
+	unsent_earnings -= high_cut
+
+#undef A1_CURVE
+#undef PROCESS_CAP
+#undef MINIMUM_BAR
+
+// Beam related procs
+
+/obj/machinery/power/transmission_laser/proc/setup_lasers()
+	/// This is why we set the range we did
+	var/turf/last_step = get_step(get_front_turf(), dir)
+	for(var/num = 1 to range + 1)
+		var/obj/effect/transmission_beam/new_beam = new(last_step)
+		new_beam.host = src
+		new_beam.dir = dir
+		laser_effects += new_beam
+
+		last_step = get_step(last_step, dir)
+
+/obj/machinery/power/transmission_laser/proc/destroy_lasers()
+	for(var/obj/effect/transmission_beam/listed_beam as anything in laser_effects)
+		laser_effects -= listed_beam
+		qdel(listed_beam)
+
+/// This is called every time something enters our beams
+/obj/machinery/power/transmission_laser/proc/atom_entered_beam(obj/effect/transmission_beam/triggered, atom/movable/potential_victim)
+	var/mw_power = (output_number * power_format_multi_output) / (1 MW)
+	if(!isliving(potential_victim))
```

Comment:
No behavior at all if an object enters the beam? Might want to try to light it on fire/give it firestacks.

Reply:
Yeah, I guess it should have been handled in the blocking mechanic, though that probably doesn't take non dense things into account anyhow. I'll add it while I'm at it.

Reply:
Sorted it